clients/common/nm-secret-agent-simple.c

```diff
@@ -595,9 +899,7 @@ request_secrets_from_ui (NMSecretAgentSimpleRequest *request)
 		                     "Cannot service a secrets request %s for a %s connection",
 		                     request->request_id,
 		                     nm_connection_get_connection_type (request->connection));
-		request->callback (NM_SECRET_AGENT_OLD (request->self), request->connection,
-		                   NULL, error, request->callback_data);
-		g_hash_table_remove (priv->requests, request->request_id);
+		nm_secret_agent_simple_request_cancel (request, error);
```

Comment:
it seems ugly, that the API sometimes invokes the callback synchronously and sometimes asynchronously.

Reply:
Doesn't seem that ugly to me.